### DISCUSSION TITLE
Resizing image with/without Detectron2 not giving equivalent results

### DISCUSSION THREAD

**Discussion by NielsRogge** [facebookresearch/detectron2#3239] [Q&A] [answered]
I'd like to transform images in a similar way to Detectron2, using only PIL and Numpy. However, the following is not giving me equivalent results. I've made a [Colab](https://colab.research.google.com/drive/1jJfyXIxRw7lqAtc2zzO6V9nfmI_CpB89?usp=sharing) to reproduce my issue.

```
# using Detectron 2
from detectron2.data.detection_utils import read_image
from detectron2.data.transforms import ResizeTransform, TransformList
import torch

def load_image_detectron2(image_path):
    image = read_image(image_path, format="BGR")
    h = image.shape[0]
    w = image.shape[1]
    img_trans = TransformList([ResizeTransform(h=h, w=w, new_h=224, new_w=224)])
    image = torch.tensor(img_trans.apply_image(image).copy()).permute(2, 0, 1)  
    return image, (w, h)
```

```
# using PIL + NumPy only
from PIL import Image
import numpy as np

def load_image(image_path):
    image = Image.open(image_path).convert("RGB")
    w, h = image.size
    image = image.resize((224, 224))
    image = np.asarray(image)
    image = image[:, :, ::-1] # flip color channels from RGB to BGR
    image = image.transpose(2, 0, 1) # move channels to first dimension
    return image, (w, h)
```

When printing the PyTorch tensor and Numpy array, they are not the same. I wonder why this might be the case. It seems to have to do with resizing (before resizing, both are equivalent). However, resizing does not involve the color channels, so are PIL.resize and ResizeTransform different?

**Comment by ppwwyyxx** [comment, marked as answer]
The default interpolation is different. Please see docs of https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.resize and https://detectron2.readthedocs.io/en/latest/modules/data_transforms.html#detectron2.data.transforms.ResizeTransform

**Reply by NielsRogge** [reply]
Thanks! This solved it.
